File: src/backend/go/todo-go-api-base/models/todo.go
package models

import "github.com/google/uuid"

// Create a new type named Todo that is a struct with fields: Id, Title, IsCompleted, InsertedAt, UpdatedAt, IsDeleted
// Todo extends Entity

type Todo struct {
	EntityBase
	Title       string    `gorm:"column:Title"`
	IsCompleted bool      `gorm:"column:IsCompleted"`
	CategoryId  uuid.UUID `gorm:"column:CategoryId"`
	Category    Category  `gorm:"foreignKey:CategoryId"`
}

File: src/backend/Java/todoapibase/src/main/java/com/congdinh/todoapibase/models/Todo.java
package com.congdinh.todoapibase.models;

import java.time.LocalDateTime;
import java.util.UUID;

import org.hibernate.validator.constraints.Length;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;

@Entity
public class Todo extends EntityBase {
    @NotBlank
    @Length(max = 100, min = 3)
    private String name;

    private Boolean isCompleted;

    protected Todo() {
    }

    public Todo(UUID id, String name, Boolean isCompleted, LocalDateTime insertedAt, LocalDateTime updatedAt,
            Boolean isDeleted) {
        super(id, insertedAt, updatedAt, isDeleted);
        this.name = name;
        this.isCompleted = isCompleted;
    }

    public String getName() {
        return name;
    }

    public Boolean getIsCompleted() {
        return isCompleted;
    }

    public void setName(String title) {
        this.name = title;
    }

    public void setIsCompleted(Boolean isCompleted) {
        this.isCompleted = isCompleted;
    }

    @ManyToOne
    private Category category;

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }
}
